sourced/ml/cmd_entries/repos2df.py

```diff
@@ -3,23 +3,30 @@
 
 from sourced.ml.extractors import create_extractors_from_args
 from sourced.ml.models import OrderedDocumentFrequencies, QuantizationLevels
-from sourced.ml.transformers import Ignition, UastExtractor, UastDeserializer, LanguageSelector, \
-    BagFeatures2DocFreq, HeadFiles, Uast2BagFeatures, Counter, Cacher, Uast2Quant, UastRow2Document
-from sourced.ml.utils import create_engine
-from sourced.ml.utils.engine import pipeline_graph, pause
+from sourced.ml.transformers import BagFeatures2DocFreq, Cacher, Counter, create_parquet_loader, \
+    HeadFiles, Ignition, LanguageSelector, UastDeserializer, UastExtractor, UastRow2Document, \
+    Uast2BagFeatures, Uast2Quant
+from sourced.ml.utils import create_engine, pause, pipeline_graph
 
 
 @pause
 def repos2df_entry(args):
     log = logging.getLogger("repos2df")
-    engine = create_engine("repos2df-%s" % uuid4(), **args.__dict__)
+    session_name = "repos2df-%s" % uuid4()
     extractors = create_extractors_from_args(args)
+    if args.parquet:
+        start_point = create_parquet_loader(session_name, **args.__dict__)
+        root = start_point
+    else:
+        engine = create_engine(session_name, **args.__dict__)
+        root = engine
```

Comment:
do you want some python magic? :)

```python
root_creator = create_parquet_loader if args.parquet else create_engine
root = root_creator(session_name, **args.__dict__)
```

3 times shorter

Also, it is really hard to read and distinguish `root` and `start_point` in the code...

Reply:
Nice :o bwill incorporate in `starting_point` function then. I agree, but root is useful for two things: `pipeline_graph` and to get the spark `session`, under a same variable name. we would have to use the engine for siva files and session, as the `starting_point`, so do more conditionning ... I think we should keep both